Add recipe for packaged version of the Alma ATM tools.

tskisner opened the pull request:
Hello conda-forge team, this is the first in a series of several recipes for tools used in analysis of data from microwave telescopes.  We are migrating our software stack to conda packages.

I tested the package build inside the anvil docker container.  The package itself also builds on windows, but there may be small tweaks needed to bld.bat.  We'll see how that goes.

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams.
Consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/libaatm```) and found some lint.

Here's what I've got...


For **recipes/libaatm**:

 * The recipe must have some tests.

tskisner commented:
Ok, I'll work on these issues soon.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/libaatm```) and found it was in an excellent condition.

marcelotrevisani commented on recipes/libaatm/build.sh:
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Could you please remove the shebang?
``conda-build`` already takes care of it! :)
```suggestion
```

marcelotrevisani commented on recipes/libaatm/build.sh:
```diff
@@ -0,0 +1,32 @@
+    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+    ${CMAKE_PLATFORM_FLAGS[@]} \
+    -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
```
Just to avoid any kind of problem, I usually like to also add
```suggestion
    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
    -DCMAKE_PREFIX_PATH="${PREFIX}" \
```

marcelotrevisani commented on recipes/libaatm/meta.yaml:
```diff
@@ -0,0 +1,40 @@
+  version: {{ version }}
+
+source:
+  fn: libaatm-{{ version }}.tar.gz
```
This line is not necessary, you can remove it
```suggestion
```

marcelotrevisani commented on recipes/libaatm/meta.yaml:
```diff
@@ -0,0 +1,40 @@
+    - cmake
+  host:
+  run:  
+
```
you need to add the ``test`` section
Please see
https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=test#test

tskisner commented:
Thanks for your earlier feedback @marcelotrevisani, I think those items are addressed.  I added an existence test in meta.yaml and the build script runs the included tests (which are not installed).  

Although the conda recipe works for me locally on a windows machine with latest visual C++, the builds don't work yet within the conda-forge infrastructure.  So I have added a windows skip = True for now.

Let me know if there are any other changes needed
